src/main/java/com/khoros/twitterapp/StatusUpdateResource.java

```diff
@@ -0,0 +1,32 @@
+package com.khoros.twitterapp;
+
+import twitter4j.TwitterFactory;
+import twitter4j.Twitter;
+import twitter4j.Status;
+import twitter4j.TwitterException;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+
+@Path("/tweet")
+@Consumes("application/x-www-form-urlencoded")
+@Produces("application/json")
+public class StatusUpdateResource {
+    @POST
+    public Response postStatus(@FormParam("message") String tweetText) {
+        Twitter factory = new TwitterFactory().getSingleton();
+        String statusText = tweetText.trim();
+        try {
+            if (statusText.length() > 0 && statusText.length() <= 280) {
+                Status newStatus = factory.updateStatus(statusText);
+                return Response.status(200).entity(newStatus).type("application/json").build();
+            } else {
+                throw new LengthException(statusText.length());
```

Comment:
instead of throwing the LengthException and catching it right after, why not just return back the 403 status here

Reply:
It provides a different causeMessage depending on the length of the message. Either "Error: Tweet text surpassed 280 characters." or "Error: No tweet text entered."

Reply:
I updated it anyway by adding a couple of if-else-statements